src/category_theory/model/wfs.lean

```diff
@@ -47,6 +49,18 @@ begin
   refine ⟨r.ib ≫ l, _, _⟩,
   { rw [←category.assoc, ←r.hi, category.assoc, hl₁, ←category.assoc, r.ha, category.id_comp] },
   { rw [category.assoc, hl₂, ←category.assoc, r.hb, category.id_comp] }
+end    
+
+lemma is_wfs.retract_right {L R :morphism_class M } (w : is_wfs L R)
```

Comment:
I ended up copying the proof of the previous lemma for this. I first tried to prove it by applying the previous lemma to the opposite category, but I quickly got bogged down.

Reply:
Yes. It was probably made more difficult by the way that the opposite category was implemented at the time. We later found that it was very easy to get confused about whether we were considering objects and morphisms in a category or in its opposite. Nowadays we have a design in which you need to apply specific functions to convert between a category and its opposite, which helps keep things straight.

Ideally we will some day have a method to automatically dualize theorems, but this doesn't exist yet.